### ISSUE TITLE
runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by sidwise** [influxdata/influxdb#23016]
System: Docker
Version: influxDB 2.1
Built from official Docker Image

When I try to load different fields from existing buckets I get the below error:

`ts=2021-12-21T15:47:17.646101Z lvl=info msg="Execute source panic" log_id=0YZQdeMG000 service=storage-reads error="panic: runtime error: invalid memory address or nil pointer dereference" stacktrace="goroutine 1639 [running]:\nruntime/debug.Stack()\n\t/home
/circleci/.tools/go/src/runtime/debug/stack.go:24 +0x65\ngithub.com/influxdata/flux/execute.(*executionState).do.func1.1()\n\t/home/circleci/go/pkg/mod/github.com/influxdata/flux@v0.139.0/execute/executor.go:286 +0x205\npanic({0x3c52600, 0x53e34d0})\n\t/home/circleci/.to
ols/go/src/runtime/panic.go:1038 +0x215\ngithub.com/influxdata/influxdb/v2/v1/services/storage.(*Store).measurementFields.func1()\n\t/home/circleci/go/src/github.com/influxdata/influxdb/v1/services/storage/store.go:546 +0x1c\npanic({0x3c7bc60, 0xc0024e7ec0})\n\t/home/cir
cleci/.tools/go/src/runtime/panic.go:1038 +0x215\ngithub.com/influxdata/influxdb/v2/v1/services/storage.(*Store).measurementFields(0xc0009b61b0, {0x40a6b18, 0xc0024ea780}, 0xc001634bd0)\n\t/home/circleci/go/src/github.com/influxdata/influxdb/v1/services/storage/store.go:549 +0x33f\ngithub.com/influxdata/influxdb/v2/v1/services/storage.(*Store).TagValues(0x4079ad0, {0x40a6b18, 0xc0024ea780}, 0xc0024e7d70)\n\t/home/circleci/go/src/github.com/influxdata/influxdb/v1/services/storage/store.go:406 +0x390\ngithub.com/influxdata/influxdb/v2/storage/flux.(*tagValuesIterator).Do(0xc0024a4e80, 0x61c1f705)\n\t/home/circleci/go/src/github.com/influxdata/influxdb/storage/flux/reader.go:970 +0x1c6\ngithub.com/influxdata/influxdb/v2/query/stdlib/influxdata/influxdb.(*Source).processTables(0xc0004695f0, {0x40a6b18, 0xc0024ea780}, {0x406f510, 0xc0024a4e80}, 0xc0020e4bb8)\n\t/home/circleci/go/src/github.com/influxdata/influxdb/query/stdlib/influxdata/influxdb/source.go:76 +0xbf\ngithub.com/influxdata/influxdb/v2/query/stdlib/influxdata/influxdb.(*readTagValuesSource).run(0xc0004695f0, {0x40a6b18, 0xc0024ea780})\n\t/home/circleci/go/src/github.com/influxdata/influxdb/query/stdlib/influxdata/influxdb/source.go:492 +0x114\ngithub.com/influxdata/influxdb/v2/query/stdlib/influxdata/influxdb.(*Source).Run(0xc0004695f0, {0x40a6b18, 0xc0024ea780})\n\t/home/circleci/go/src/github.com/influxdata/influxdb/query/stdlib/influxdata/influxdb/source.go:56 +0x168\ngithub.com/influxdata/flux/execute.(*executionState).do.func1({0x40a86e0, 0xc0004695f0})\n\t/home/circleci/go/pkg/mod/github.com/influxdata/flux@v0.139.0/execute/executor.go
:291 +0x20e\ncreated by github.com/influxdata/flux/execute.(*executionState).do\n\t/home/circleci/go/pkg/mod/github.com/influxdata/flux@v0.139.0/execute/executor.go:261 +0x228\n"`

![image](https://user-images.githubusercontent.com/4213032/146960808-55a329ec-7a03-4a47-9a30-c5f52b1f2ddc.png)

**Comment by bcare-bergia** [comment]
Hello,
system: docker (+ docker-compose)
Version: influxdb:2.1 (644e7014ccc4)
Exact same problem as @sidwise 

Note that the same server is queried by a Grafana instance, and from Grafana the data can be queried without problem from the same bucket. The Influxdb dashboard was also working fine before we started seeing this error.

lvl=error msg="a panic has occurred" log_id=0ZelezMG000 handler=panic error="/api/v2/query?org=ekobot: runtime error: invalid memory address or nil pointer dereference" stacktrace="
goroutine 1388 [running]:
runtime/debug.Stack()
    /home/circleci/.tools/go/src/runtime/debug/stack.go:24 +0x65
github.com/influxdata/influxdb/v2/http.baseHandler.panic({{0x4058600, 0xc000ac9620}}, {0x4077400, 0xc000f3ec00}, 0xc00091fb90, {0x3c52600, 0x53e34d0})
    /home/circleci/go/src/github.com/influxdata/influxdb/http/router.go:96 +0x1b3
github.com/influxdata/httprouter.(*Router).recv(0xc000bee1b0, {0x4077400, 0xc000f3ec00}, 0xc001c49b00)
    /home/circleci/go/pkg/mod/github.com/influxdata/httprouter@v1.3.1-0.20191122104820-ee83e2772f69/router.go:361 +0x5e
panic({0x3c52600, 0x53e34d0})
    /home/circleci/.tools/go/src/runtime/panic.go:1038 +0x215
github.com/influxdata/influxdb/v2/tsdb/engine/tsm1.(*KeyCursor).ReadFloatArrayBlock(0x0, 0xc001babb60)
    /home/circleci/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/file_store_array.gen.go:13 +0x35
github.com/influxdata/influxdb/v2/tsdb/engine/tsm1.(*floatArrayAscendingCursor).nextTSM(0xc0012f6910)
    /home/circleci/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/array_cursor.gen.go:155 +0x37
github.com/influxdata/influxdb/v2/tsdb/engine/tsm1.(*floatArrayAscendingCursor).Next(0xc0012f6910)
    /home/circleci/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/array_cursor.gen.go:114 +0x352
github.com/influxdata/influxdb/v2/storage/reads.(*floatMultiShardArrayCursor).Next(0xc0013efbc0)
    /home/circleci/go/src/github.com/influxdata/influxdb/storage/reads/array_cursor.gen.go:297 +0x30
github.com/influxdata/influxdb/v2/storage/flux.(*floatTable).advance(0xc000f00f70)
    /home/circleci/go/src/github.com/influxdata/influxdb/storage/flux/table.gen.go:88 +0x34
github.com/influxdata/influxdb/v2/storage/flux.(*table).do(0xc000dc1e88, 0xc0015e4900, 0xc000dc1e78)
    /home/circleci/go/src/github.com/influxdata/influxdb/storage/flux/table.go:95 +0x1ec
github.com/influxdata/influxdb/v2/storage/flux.(*floatTable).Do(0xc0013d2680, 0x3)
    /home/circleci/go/src/github.com/influxdata/influxdb/storage/flux/table.gen.go:84 +0x37
github.com/influxdata/flux/csv.(*ResultEncoder).Encode.func1({0x7f419eecdae8, 0xc000f00f70})
    /home/circleci/go/pkg/mod/github.com/influxdata/flux@v0.139.0/csv/result.go:954 +0x5b7
github.com/influxdata/flux/execute.(*result).Do(0xc001575740, 0xc001169ea0)
    /home/circleci/go/pkg/mod/github.com/influxdata/flux@v0.139.0/execute/result.go:71 +0xe2
github.com/influxdata/influxdb/v2/query/control.(*errorCollectingTableIterator).Do(0xc00116a7b0, 0xc00165fc28)
    /home/circleci/go/src/github.com/influxdata/influxdb/query/control/controller.go:989 +0x29
github.com/influxdata/flux/csv.(*ResultEncoder).Encode(0xc00192f650, {0x404f9c0, 0xc001d41518}, {0x406f218, 0xc00116a6d8})
    /home/circleci/go/pkg/mod/github.com/influxdata/flux@v0.139.0/csv/result.go:901 +0x3d8
github.com/influxdata/flux.(*DelimitedMultiResultEncoder).Encode(0xc00192f680, {0x404f9c0, 0xc001d41440}, {0x40b2878, 0xc00192f620})
    /home/circleci/go/pkg/mod/github.com/influxdata/flux@v0.139.0/result.go:295 +0xcc
github.com/influxdata/influxdb/v2/query.ProxyQueryServiceAsyncBridge.Query({{0x4052760, 0xc0011e2100}}, {0x40a6bc0, 0xc00192f3b0}, {0x404f9c0, 0xc001d41440}, 0xc000ef51f0)
    /home/circleci/go/src/github.com/influxdata/influxdb/query/bridges.go:146 +0x238
github.com/influxdata/influxdb/v2/http.routingQueryService.Query({{0x4073660, 0xc000df0680}, {0x4073660, 0xc000df0680}}, {0x40a6bc0, 0xc00192f3b0}, {0x404f9c0, 0xc001d41440}, 0xc000ef51f0)
    /home/circleci/go/src/github.com/influxdata/influxdb/http/query_handler.go:694 +0x291
github.com/influxdata/influxdb/v2/http.(*FluxHandler).handleQuery(0xc000eef680, {0x4074580, 0xc001c49b00}, 0xc000dc2c08)
    /home/circleci/go/src/github.com/influxdata/influxdb/http/query_handler.go:204 +0x949
net/http.HandlerFunc.ServeHTTP(0x3b69c00, {0x4074580, 0xc001c49b00}, 0x4)
    /home/circleci/.tools/go/src/net/http/server.go:2046 +0x2f
github.com/NYTimes/gziphandler.GzipHandlerWithOpts.func1.1({0x4077400, 0xc000f3ec00}, 0x90)
    /home/circleci/go/pkg/mod/github.com/!n!y!times/gziphandler@v1.0.1/gzip.go:289 +0x262
net/http.HandlerFunc.ServeHTTP(0xc000b4d440, {0x4077400, 0xc000f3ec00}, 0x7f41cd36f5b8)
    /home/circleci/.tools/go/src/net/http/server.go:2046 +0x2f
github.com/influxdata/influxdb/v2/http.(*proxyHandler).ServeHTTP(0xc000f2d040, {0x4077400, 0xc000f3ec00}, 0x90)
    /home/circleci/go/src/github.com/influxdata/influxdb/http/proxy_handler.go:33 +0x68
github.com/influxdata/httprouter.(*Router).Handler.func1({0x4077400, 0xc000f3ec00}, 0x3c4d440, {0x0, 0x3b69c00, 0xc0008ab8d0})
    /home/circleci/go/pkg/mod/github.com/influxdata/httprouter@v1.3.1-0.20191122104820-ee83e2772f69/router.go:325 +0x2c4
github.com/influxdata/httprouter.(*Router).ServeHTTP(0xc000bee1b0, {0x4077400, 0xc000f3ec00}, 0xc0019c9a00)
    /home/circleci/go/pkg/mod/github.com/influxdata/httprouter@v1.3.1-0.20191122104820-ee83e2772f69/router.go:453 +0x4c3
github.com/go-chi/chi.(*Mux).Mount.func1({0x4077400, 0xc000f3ec00}, 0xc0019c9a00)
    /home/circleci/go/pkg/mod/github.com/go-chi/chi@v4.1.0+incompatible/mux.go:298 +0x13b
net/http.HandlerFunc.ServeHTTP(0x3c2d5a0, {0x4077400, 0xc000f3ec00}, 0xc001061160)
    /home/circleci/.tools/go/src/net/http/server.go:2046 +0x2f
github.com/go-chi/chi.(*Mux).routeHTTP(0xc000be4360, {0x4077400, 0xc000f3ec00}, 0xc0019c9a00)
    /home/circleci/go/pkg/mod/github.com/go-chi/chi@v4.1.0+incompatible/mux.go:431 +0x1f9
net/http.HandlerFunc.ServeHTTP(0xc00192ed50, {0x4077400, 0xc000f3ec00}, 0xc000dc30b8)
    /home/circleci/.tools/go/src/net/http/server.go:2046 +0x2f
github.com/influxdata/influxdb/v2/kit/transport/http.SetCORS.func1({0x4077400, 0xc000f3ec00}, 0xc0019c9a00)
    /home/circleci/go/src/github.com/influxdata/influxdb/kit/transport/http/middleware.go:36 +0x308
net/http.HandlerFunc.ServeHTTP(0x0, {0x4077400, 0xc000f3ec00}, 0x203000)
    /home/circleci/.tools/go/src/net/http/server.go:2046 +0x2f
github.com/go-chi/chi/middleware.StripSlashes.func1({0x4077400, 0xc000f3ec00}, 0xc0019c9a00)
    /home/circleci/go/pkg/mod/github.com/go-chi/chi@v4.1.0+incompatible/middleware/strip.go:25 +0x102
net/http.HandlerFunc.ServeHTTP(0xc00192ed50, {0x4077400, 0xc000f3ec00}, 0x30)
    /home/circleci/.tools/go/src/net/http/server.go:2046 +0x2f
github.com/influxdata/influxdb/v2/kit/transport/http.SkipOptions.func1({0x4077400, 0xc000f3ec00}, 0xc0019c9a00)
    /home/circleci/go/src/github.com/influxdata/influxdb/kit/transport/http/middleware.go:81 +0xb9
net/http.HandlerFunc.ServeHTTP(0xc00192edb0, {0x4077400, 0xc000f3ec00}, 0xc000dc3238)
    /home/circleci/.tools/go/src/net/http/server.go:2046 +0x2f
github.com/influxdata/influxdb/v2/http.panicMW.func1.1({0x4077400, 0xc000f3ec00}, 0xc000dc32f0)
    /home/circleci/go/src/github.com/influxdata/influxdb/http/router.go:126 +0x9b
net/http.HandlerFunc.ServeHTTP(0x90, {0x4077400, 0xc000f3ec00}, 0x100000002394ddb)
    /home/circleci/.tools/go/src/net/http/server.go:2046 +0x2f
github.com/go-chi/chi.(*Mux).ServeHTTP(0xc000be4360, {0x4077400, 0xc000f3ec00}, 0xc0019c9a00)
    /home/circleci/go/pkg/mod/github.com/go-chi/chi@v4.1.0+incompatible/mux.go:70 +0x40e
github.com/influxdata/influxdb/v2/kit/feature.(*Handler).ServeHTTP(0xc000da2f40, {0x4077400, 0xc000f3ec00}, 0xc0019c9900)
    /home/circleci/go/src/github.com/influxdata/influxdb/kit/feature/middleware.go:43 +0x3c2
github.com/influxdata/influxdb/v2/http.(*AuthenticationHandler).ServeHTTP(0xc001231570, {0x4077400, 0xc000f3ec00}, 0xc0019c9800)
    /home/circleci/go/src/github.com/influxdata/influxdb/http/authentication_middleware.go:127 +0x467
github.com/influxdata/influxdb/v2/kit/transport/http.SetCORS.func1({0x4077400, 0xc000f3ec00}, 0xc0019c9800)
    /home/circleci/go/src/github.com/influxdata/influxdb/kit/transport/http/middleware.go:36 +0x308
net/http.HandlerFunc.ServeHTTP(0xc00192ed50, {0x4077400, 0xc000f3ec00}, 0xc001061161)
    /home/circleci/.tools/go/src/net/http/server.go:2046 +0x2f
github.com/influxdata/influxdb/v2/kit/transport/http.SkipOptions.func1({0x4077400, 0xc000f3ec00}, 0xc0019c9800)
    /home/circleci/go/src/github.com/influxdata/influxdb/kit/transport/http/middleware.go:81 +0xb9
net/http.HandlerFunc.ServeHTTP(0x33d8a90, {0x4077400, 0xc000f3ec00}, 0xc)
    /home/circleci/.tools/go/src/net/http/server.go:2046 +0x2f
github.com/influxdata/influxdb/v2/http.(*PlatformHandler).ServeHTTP(0xc000da3040, {0x4077400, 0xc000f3ec00}, 0xc0019c9800)
    /home/circleci/go/src/github.com/influxdata/influxdb/http/platform_handler.go:81 +0x1d2
github.com/go-chi/chi.(*Mux).Mount.func1({0x4077400, 0xc000f3ec00}, 0xc0019c9800)
    /home/circleci/go/pkg/mod/github.com/go-chi/chi@v4.1.0+incompatible/mux.go:298 +0x13b
net/http.HandlerFunc.ServeHTTP(0x203000, {0x4077400, 0xc000f3ec00}, 0xc000f3ebe0)
    /home/circleci/.tools/go/src/net/http/server.go:2046 +0x2f
github.com/influxdata/influxdb/v2/kit/transport/http.Metrics.func1.1({0x407c1d0, 0xc00086ddc0}, 0xc0019c9800)
    /home/circleci/go/src/github.com/influxdata/influxdb/kit/transport/http/middleware.go:66 +0x1e2
net/http.HandlerFunc.ServeHTTP(0xc000dc3830, {0x407c1d0, 0xc00086ddc0}, 0x0)
    /home/circleci/.tools/go/src/net/http/server.go:2046 +0x2f
github.com/influxdata/influxdb/v2/kit/transport/http.Trace.func1.1({0x407c1d0, 0xc00086ddc0}, 0x1)
    /home/circleci/go/src/github.com/influxdata/influxdb/kit/transport/http/middleware.go:106 +0x30c
net/http.HandlerFunc.ServeHTTP(0xc000daa3c0, {0x407c1d0, 0xc00086ddc0}, 0xc00162d0f8)
    /home/circleci/.tools/go/src/net/http/server.go:2046 +0x2f
github.com/go-chi/chi.(*ChainHandler).ServeHTTP(0x3c2d5a0, {0x407c1d0, 0xc00086ddc0}, 0xc001276bd5)
    /home/circleci/go/pkg/mod/github.com/go-chi/chi@v4.1.0+incompatible/chain.go:31 +0x2c
github.com/go-chi/chi.(*Mux).routeHTTP(0xc000d0ecc0, {0x407c1d0, 0xc00086ddc0}, 0xc0019c9700)
    /home/circleci/go/pkg/mod/github.com/go-chi/chi@v4.1.0+incompatible/mux.go:431 +0x1f9
net/http.HandlerFunc.ServeHTTP(0x88, {0x407c1d0, 0xc00086ddc0}, 0x90)
    /home/circleci/.tools/go/src/net/http/server.go:2046 +0x2f
github.com/influxdata/influxdb/v2/http.(*AddHeader).Middleware.func1({0x407c1d0, 0xc00086ddc0}, 0x53c4e01)
    /home/circleci/go/src/github.com/influxdata/influxdb/http/handler.go:102 +0x77
net/http.HandlerFunc.ServeHTTP(0x40a6b18, {0x407c1d0, 0xc00086ddc0}, 0x53c4e90)
    /home/circleci/.tools/go/src/net/http/server.go:2046 +0x2f
github.com/go-chi/chi.(*Mux).ServeHTTP(0xc000d0ecc0, {0x407c1d0, 0xc00086ddc0}, 0xc0019c9600)
    /home/circleci/go/pkg/mod/github.com/go-chi/chi@v4.1.0+incompatible/mux.go:86 +0x3c2
github.com/influxdata/influxdb/v2/http.(*Handler).ServeHTTP(0xc001276be3, {0x407c1d0, 0xc00086ddc0}, 0x0)
    /home/circleci/go/src/github.com/influxdata/influxdb/http/handler.go:166 +0x2f
net/http.serverHandler.ServeHTTP({0x4072558}, {0x407c1d0, 0xc00086ddc0}, 0xc0019c9600)
    /home/circleci/.tools/go/src/net/http/server.go:2878 +0x43b
net/http.(*conn).serve(0xc001654e60, {0x40a6bc0, 0xc000e163f0})
    /home/circleci/.tools/go/src/net/http/server.go:1929 +0xb08
created by net/http.(*Server).Serve    /home/circleci/.tools/go/src/net/http/server.go:3033 +0x4e8"

**Comment by pelerner** [comment]
Same here .... 

System Raspi 4b 8GB, aarch64
InfluxDB v2.1.1, package provided by arch linux arm project.
Writing data from openhab v3.2 to influxdb v2.1.1.

Error Message on Data explorer: "Failed to load tag values for ...field" on _all_ measurements.

Restarting InfluxDB does not help.
DB seems to be corrupted.

```
Feb 14 20:19:22 openhab3 influxd[341]: 
ts=2022-02-14T19:19:22.293605Z 
lvl=info msg="Execute source panic" 
log_id=0ZcnlK_G000 
service=storage-reads 
error="panic: runtime error: invalid memory address or nil pointer dereference" 
stacktrace=

"goroutine 271374 [running]:runtime/debug.Stack()
\truntime/debug/stack.go:24 +0x88
github.com/influxdata/flux/execute.(*executionState).do.func1.1(0x4000b6e7e0)
\tgithub.com/influxdata/flux@v0.139.0/execute/executor.go:286 +0x230
panic({0xaaaae5335680, 0xaaaae69fc4f0})
\truntime/panic.go:1038 +0x21c
github.com/influxdata/influxdb/v2/v1/services/storage.(*Store).measurementFields.func1({0x0, 0x0})
\tgithub.com/influxdata/influxdb/v2/v1/services/storage/store.go:546 +0x20
panic({0xaaaae535e7a0, 0x400151d5c0})
\truntime/panic.go:1038 +0x21c
github.com/influxdata/influxdb/v2/v1/services/storage.(*Store).measurementFields(0x40037f8720, {0xaaaae57713b8, 0x40016b6f80}, 0x4000077b98)
\tgithub.com/influxdata/influxdb/v2/v1/services/storage/store.go:549 +0x304
github.com/influxdata/influxdb/v2/v1/services/storage.(*Store).TagValues(0x40037f8720, {0xaaaae57713b8, 0x40016b6f80}, 0x400151d470)
\tgithub.com/influxdata/influxdb/v2/v1/services/storage/store.go:406 +0x390
github.com/influxdata/influxdb/v2/storage/flux.(*tagValuesIterator).Do(0x40014c3080, 0x4001518c20)
\tgithub.com/influxdata/influxdb/v2/storage/flux/reader.go:970 +0x204
github.com/influxdata/influxdb/v2/query/stdlib/influxdata/influxdb.(*Source).processTables(0x4000aaaf70, {0xaaaae57713b8, 0x40016b6f80}, {0xaaaae573d810, 0x40014c3080}, 0x16d3bd3ddb7cfc6e)
\tgithub.com/influxdata/influxdb/v2/query/stdlib/influxdata/influxdb/source.go:76 +0xa8
github.com/influxdata/influxdb/v2/query/stdlib/influxdata/influxdb.(*readTagValuesSource).run(0x4000aaaf70, {0xaaaae57713b8, 0x40016b6f80})
\tgithub.com/influxdata/influxdb/v2/query/stdlib/influxdata/influxdb/source.go:492 +0x104
github.com/influxdata/influxdb/v2/query/stdlib/influxdata/influxdb.(*Source).Run(0x4000aaaf70, {0xaaaae57713b8, 0x40016b6f80})
\tgithub.com/influxdata/influxdb/v2/query/stdlib/influxdata/influxdb/source.go:56 +0x190
github.com/influxdata/flux/execute.(*executionState).do.func1(0x4000b6e7e0, 0x40025cd230, {0xaaaae5772f80, 0x4000aaaf70})
\tgithub.com/influxdata/flux@v0.139.0/execute/executor.go:291 +0x180
created by github.com/influxdata/flux/execute.(*executionState).do
\tgithub.com/influxdata/flux@v0.139.0/execute/executor.go:261 +0x158"

```

... further on ... 

```
Feb 14 08:39:49 openhab3 influxd[341]: 
ts=2022-02-14T07:39:49.300835Z 
lvl=info msg="Execute source panic" 
log_id=0ZcnlK_G000 
service=storage-reads 
error="panic: runtime error: invalid memory address or nil pointer dereference" 
stacktrace=

"goroutine 205500 [running]:
runtime/debug.Stack()
\truntime/debug/stack.go:24 +0x88
github.com/influxdata/flux/execute.(*executionState).do.func1.1(0x4000e6f680)
\tgithub.com/influxdata/flux@v0.139.0/execute/executor.go:286 +0x230
panic({0xaaaae5335680, 0xaaaae69fc4f0})
\truntime/panic.go:1038 +0x21c
github.com/influxdata/influxdb/v2/v1/services/storage.(*Store).measurementFields.func1({0x0, 0x0})
\tgithub.com/influxdata/influxdb/v2/v1/services/storage/store.go:546 +0x20
panic({0xaaaae535e7a0, 0x400099e000})
\truntime/panic.go:1038 +0x21c
github.com/influxdata/influxdb/v2/v1/services/storage.(*Store).measurementFields(0x40037f8720, {0xaaaae57713b8, 0x40018ea880}, 0x40008b6b98)
\tgithub.com/influxdata/influxdb/v2/v1/services/storage/store.go:549 +0x304
github.com/influxdata/influxdb/v2/v1/services/storage.(*Store).TagValues(0x40037f8720, {0xaaaae57713b8, 0x40018ea880}, 0x4000cf8ab0)
\tgithub.com/influxdata/influxdb/v2/v1/services/storage/store.go:406 +0x390
github.com/influxdata/influxdb/v2/storage/flux.(*tagValuesIterator).Do(0x4000e48600, 0x4000e02940)
\tgithub.com/influxdata/influxdb/v2/storage/flux/reader.go:970 +0x204
github.com/influxdata/influxdb/v2/query/stdlib/influxdata/influxdb.(*Source).processTables(0x400051d6c0, {0xaaaae57713b8, 0x40018ea880}, {0xaaaae573d810, 0x4000e48600}, 0x16d3971141f1c8f9)
\tgithub.com/influxdata/influxdb/v2/query/stdlib/influxdata/influxdb/source.go:76 +0xa8
github.com/influxdata/influxdb/v2/query/stdlib/influxdata/influxdb.(*readTagValuesSource).run(0x400051d6c0, {0xaaaae57713b8, 0x40018ea880})
\tgithub.com/influxdata/influxdb/v2/query/stdlib/influxdata/influxdb/source.go:492 +0x104
github.com/influxdata/influxdb/v2/query/stdlib/influxdata/influxdb.(*Source).Run(0x400051d6c0, {0xaaaae57713b8, 0x40018ea880})
\tgithub.com/influxdata/influxdb/v2/query/stdlib/influxdata/influxdb/source.go:56 +0x190
github.com/influxdata/flux/execute.(*executionState).do.func1(0x4000e6f680, 0x40013aa9c0, {0xaaaae5772f80, 0x400051d6c0})
\tgithub.com/influxdata/flux@v0.139.0/execute/executor.go:291 +0x180
created by github.com/influxdata/flux/execute.(*executionState).do
\tgithub.com/influxdata/flux@v0.139.0/execute/executor.go:261 +0x158"
```

**Comment by jufis** [comment]
how did you solve this? delete all data and start from scratch?

**Comment by davidby-influx** [comment]
[These panics](https://github.com/influxdata/influxdb/issues/23016#issuecomment-1039479819) and [this panic](https://github.com/influxdata/influxdb/issues/23016#issue-1085982770) have been fixed: [fix: handle a potential nil iterator leading to a panic](https://github.com/influxdata/influxdb/pull/23520)

[This](https://github.com/influxdata/influxdb/issues/23016#issuecomment-1038740267) is a different panic.  Please open a new issue for it @bcare-bergia if you still encounter it with v2.7.12
